src/stm32f0xx/gpio.c

```diff
@@ -7,44 +7,33 @@
 #include "stm32f0xx_gpio.h"
 #include "stm32f0xx_rcc.h"
 
-static GPIO_TypeDef *gpio_port_map[] = {GPIOA, GPIOB, GPIOC, GPIOD, GPIOE, GPIOF};
-static uint32_t gpio_rcc_ahb_timer_map[] = {RCC_AHBPeriph_GPIOA, RCC_AHBPeriph_GPIOB,
-                                            RCC_AHBPeriph_GPIOC, RCC_AHBPeriph_GPIOD,
-                                            RCC_AHBPeriph_GPIOE, RCC_AHBPeriph_GPIOF};
+static GPIO_TypeDef *gpio_port_map[] = { GPIOA, GPIOB, GPIOC, GPIOD, GPIOE, GPIOF };
+static uint32_t gpio_rcc_ahb_timer_map[] = { RCC_AHBPeriph_GPIOA, RCC_AHBPeriph_GPIOB,
+                                             RCC_AHBPeriph_GPIOC, RCC_AHBPeriph_GPIOD,
+                                             RCC_AHBPeriph_GPIOE, RCC_AHBPeriph_GPIOF };
 
-#define NUM_PINS 16
-#define NUM_PORTS sizeof(gpio_port_map) / sizeof(GPIO_TypeDef)
+#define GPIO_NUM_PINS 16
+#define GPIO_NUM_PORTS 6
 
 // Determines if an GPIOAddress is valid based on the defined number of ports and pins.
-static inline bool prv_is_address_valid(const GPIOAddress *address) {
-  if (address->port >= NUM_PORTS || address->pin >= NUM_PINS) {
-    return false;
-  }
-  return true;
+static bool prv_is_address_valid(const GPIOAddress *address) {
+  return !(address->port >= GPIO_NUM_PORTS || address->pin >= GPIO_NUM_PINS);
 }
 
-// TODO(ckitagawa): Consider moving these two inline functions to the header as they will be used
-// more or less universally between the implementations.
+// TODO(ckitagawa): Consider moving these two functions to the header as they will be used more or
+// less universally between the implementations.
 
 // Determines if a GPIOState is valid based on the enums.
-static inline bool prv_is_state_valid(const GPIOState *state) {
-  if (*state >= NUM_GPIO_STATE) {
-    return false;
-  }
-  return true;
-}
+static bool prv_is_state_valid(const GPIOState *state) { return !(*state >= NUM_GPIO_STATE); }
```

Comment:
I'm not a huge fan of single-line functions where the function prototype and body are on the same line. I think it can get a little messy and hard to understand.

Based on the function name, I think it would make more sense to check within bounds instead of the double-negative. :)